File: app/data/repository/UserRepository.java
package data.repository;

import data.entity.User;
import data.models.FKLoginModel;
import data.models.FKSignUpModel;

public class UserRepository {

    public User signUp(User user){
        return new FKSignUpModel().createUser(user);
    }

    public User login(User user){
        return new FKLoginModel().createUser(user);
    }
}

File: app/domain/repositories/UserRepository.scala
package domain.repositories

import data.entity.User

trait UserRepository {

  def login(email: String, password: String): User

  /**
    * Creates a user
    *
    * @param username
    * @param email
    * @param password
    * @return Created user
    * @throws IllegalArgumentException if user already exists
    */
  def signUp(username: String, email: String, password: String): User
}
